run_evaluation.py

```diff
@@ -0,0 +1,52 @@
+# %% Load Annotations into dataframes
+import hourglass
+import imp
+imp.reload(hourglass)
+from hourglass import HourglassNet
+import data_generator
+imp.reload(data_generator)
+from data_generator import DataGenerator
+import evaluation
+imp.reload(evaluation)
+from evaluation import Evaluation
+from constants import *
+
+h = HourglassNet(NUM_COCO_KEYPOINTS,DEFAULT_NUM_HG,INPUT_CHANNELS,INPUT_DIM,OUTPUT_DIM)
+train_df, val_df = h.load_and_filter_annotations(DEFAULT_TRAIN_ANNOT_PATH,DEFAULT_VAL_ANNOT_PATH,0.1)
+
+# %% Declare evaluation class instance
+eval = Evaluation(
+    # ensure model_json and weights files exist in current directory and num_hg_blocks matches model_json
+    model_json='hpe_hourglass_stacks_04_batchsize_012.json',
+    weights='hpe_epoch71_val_loss_0.0417_train_loss_0.0163.hdf5',
+    df=val_df,
+    num_hg_blocks=DEFAULT_NUM_HG,
+    batch_size=1)
+print("Created Evaluation instance")
+
+# %% Save stacked evaluation heatmaps
+generator = DataGenerator(
+            df=val_df,
+            base_dir=DEFAULT_TRAIN_IMG_PATH,
+            input_dim=INPUT_DIM,
+            output_dim=OUTPUT_DIM,
+            num_hg_blocks=DEFAULT_NUM_HG,
+            shuffle=False,  
+            batch_size=1,
+            online_fetch=False)
+
+# Select image to predict heatmaps
+X_batch, y_stacked = generator[168] # choose one image for evaluation
+y_batch = y_stacked[0] # take first hourglass section
+X, y = X_batch[0], y_batch[0] # take first example of batch
+
+predict_heatmaps=eval.predict_heatmaps(h, X)
+print("Received predicted heatmaps")
+stacked_predict_heatmaps=eval.save_stacked_predict_heatmaps(predict_heatmaps)
+print("Saved predicted heatmaps")
+stacked_ground_truth_heatmaps=eval.save_stacked_ground_truth_heatmaps(X, y)
+print("Saved ground truth heatmaps")
+eval.save_stacked_evaluation_heatmaps(stacked_predict_heatmaps, stacked_ground_truth_heatmaps)
+print("Saved stacked evaluation heatmaps")
```

Comment:
Is it possible to generate the fully stacked ones without requiring the other ones to be saved to disk? A few reasons, 1) mainly so it's faster to run as everything is kept in memory, and 2) I read that Google Colab has resource limits on how much you can write to and from Google Drive. They penalize small file read/writes more than big ones.

Reply:
I doubt that this would trigger the resource limits, but if we're running a script that generates it for a ton of epochs and a ton of different model training sessions, we might want to play it safe.

Reply:
I'll check on this! I tried using the numpy arrays before and ran into some issues but I'll double check cause it would be much faster 👍

Reply:
if you get all of the hg predictions in a list or np array, you can use `np.hstack([list here])` :)

Reply:
Yeah np.hstack and np.vstack work great for the output heatmaps, but the ground truth heatmaps are using https://github.com/LinShanify/HeatMap so I'm trying to see if I can convert them to np arrays!

Reply:
Sweet! There might be a method in one of the image processing libraries that act as a file buffer to write pictures to. Worst case, since they don't change between epochs/layers, we could generate it once and use everywhere too.

Reply:
I agree with @robertklee  about only saving the final fully stacked image to disk for simplicity and efficiency. @nicolepeverley if the HeatMap submodule is feeling restrictive, keep in mind we have the option to customize it, so we could add a flag to the plot method that just returns the raw NumPy array instead of trying to plot it (just one idea).

Reply:
It looks like you were considering this already